src/flexalloc_xnvme_env.c

```diff
@@ -151,6 +189,101 @@ fla_xne_sync_seq_w_naddrs(struct xnvme_dev * dev, const uint64_t slba, const uin
   return err;
 }
 
+// Read or write out a larger chunk striped across multiple objects
+int
+fla_xne_sync_strp_seq_x(struct xnvme_dev *dev, const uint64_t offset, uint64_t nbytes,
+                        void const *buf, uint32_t strp_num, uint32_t strp_sz, bool write)
+{
+  uint32_t nsid = xnvme_dev_get_nsid(dev);
+  struct xnvme_queue *queue = NULL;
+  int err;
+  struct fla_async_cb_args cb_args = { 0 };
+  const struct xnvme_geo *geo = xnvme_dev_get_geo(dev);
+  uint32_t strp_blks = (strp_sz / geo->lba_nbytes) - 1;
+  uint64_t bytes_xfer = 0, strp_offset = offset;
+  char *strp_buf = (char *)buf;
+
+  // Nbytes must be a multiple of strp_num * strp_sz
+  if (nbytes % (strp_num * strp_sz))
+  {
+    err = -1;
+    FLA_ERR(err, "Striped write must be a multiple of strp_sz and nbytes");
+    goto exit;
+  }
+
+  // Make sure offset aligns on strp boundary
+  if (offset % strp_sz)
+  {
+    err = -1;
+    FLA_ERR(err, "Striped write offset must be aligned to strp sz");
+    goto exit;
+  }
+
+  if (nbytes/strp_num < strp_sz)
+  {
+    err = -1;
+    FLA_ERR(err, "Num bytes not large enough for strp_sz * strp_num");
+    goto exit;
+  }
+
+  err = xnvme_queue_init(dev, strp_num, 0, &queue);
+  if (FLA_ERR(err, "xnvme_queue_init"))
+    goto exit;
+
+  xnvme_queue_set_cb(queue, fla_async_cb, &cb_args);
+
+  while (bytes_xfer != nbytes)
+  {
+
+    for (uint32_t strp=0; strp < strp_num;)
+    {
+      struct xnvme_cmd_ctx *ctx = xnvme_queue_get_cmd_ctx(queue);
+      uint64_t slba = (strp_offset/geo->lba_nbytes) + (strp * geo->nsect);
```

Comment:
@Joelgranados this line is problem for conventional drives since geo->nsect is equal to zone size, which is the entire conventional device size. Do you think we should try to make this feature agnostic of device type or keep it for zoned devices only? I don't think the code is very useful for conventional drives at this point.

Reply:
NM I fixed this up.